lang/java/reef-tests/src/test/java/org/apache/reef/tests/yarnpreemption/YarnPreemptionTest.java

```diff
@@ -0,0 +1,133 @@
+            .set(DriverConfiguration.ON_EVALUATOR_ALLOCATED,
+                YarnPreemptionTestPreempteeDriver.EvaluatorAllocatedHandler.class)
+            .set(DriverConfiguration.ON_EVALUATOR_PREEMPTED,
+                YarnPreemptionTestPreempteeDriver.EvaluatorPreemptedHandler.class)
+            .set(DriverConfiguration.ON_EVALUATOR_FAILED,
+                YarnPreemptionTestPreempteeDriver.EvaluatorFailedHandler.class)
+            .build();
+
+
+        final Configuration mergedDriverConfiguration = Tang.Factory.getTang()
+            .newConfigurationBuilder(driverConfiguration, testConfigurationB).build();
+
+        try {
+          final LauncherStatus state = DriverLauncher.getLauncher(runtimeConfiguration)
+              .run(mergedDriverConfiguration, testEnvironment.getTestTimeout());
+          Assert.assertTrue("Job B (preemptee) state after execution: " + state, state.isDone());
+
+        } catch (InjectionException e) {
+          e.printStackTrace();
+        }
+      }
+    };
+
+    Thread preemptorThread = new Thread() {
+      public void run() {
+
+        try {
+          Thread.sleep(6000);
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+
+        final Configuration testConfigurationA = YarnPreemptionTestConfiguration.CONF
+            .set(YarnPreemptionTestConfiguration.JOB_QUEUE, "A")
+            .build();
+
+        final Configuration driverConfiguration = DriverConfiguration.CONF
+            .set(DriverConfiguration.GLOBAL_LIBRARIES, EnvironmentUtils.getClassLocation(this.getClass()))
+            .set(DriverConfiguration.DRIVER_IDENTIFIER, "TEST_YarnPreemptionTest_Preemptor")
+            .set(DriverConfiguration.ON_DRIVER_STARTED, YarnPreemptionTestPreemptorDriver.StartHandler.class)
+            .set(DriverConfiguration.ON_EVALUATOR_ALLOCATED,
+                YarnPreemptionTestPreemptorDriver.EvaluatorAllocatedHandler.class)
+            .build();
+
+        final Configuration mergedDriverConfiguration = Tang.Factory.getTang()
+            .newConfigurationBuilder(driverConfiguration, testConfigurationA).build();
+
+        try {
+          final LauncherStatus state = DriverLauncher.getLauncher(runtimeConfiguration)
+              .run(mergedDriverConfiguration, testEnvironment.getTestTimeout());
+          Assert.assertTrue("Job A (preemptor) state after execution: " + state, state.isSuccess());
+        } catch (InjectionException e) {
+          e.printStackTrace();
+        }
+
+      }
+    };
+
+    preempteeThread.start();
+    preemptorThread.start();
+
+    preempteeThread.join();
+    preemptorThread.join();
+  }
+
+  @Test
+  public void testYarnPreemption() throws InjectionException, InterruptedException {
+    Assume.assumeTrue("This test requires a YARN Resource Manager to connect to",
+        Boolean.parseBoolean(System.getenv("REEF_TEST_YARN")));
+    runYarnPreemptionTest();
```

Comment:
I think we can get by without an extra method - i.e. just put a body of `runYarnPreemptionTest()` here. It will be more readable and also make the `Assume` statement more visible.

Reply:
Thank you! I will address it on the next PR.